File: bindings/csharp/FastParse/FastParseFormat.cs
namespace FastParse;

/// <summary>
/// Output format produced by the native FastParse parser.
/// </summary>
public enum FastParseFormat
{
    /// <summary>Structured JSON AST output.</summary>
    Json = 1,

    /// <summary>Flat CSV rows for tabular pipelines.</summary>
    Csv = 2,

    /// <summary>Count nodes without copying output bytes.</summary>
    Stats = 3,

    /// <summary>Binary MessagePack AST output.</summary>
    Binary = 4
}

File: bindings/java/fastparse/src/main/java/dev/fastparse/FastParseFormat.java
package dev.fastparse;

/** Output format produced by FastParse. */
public enum FastParseFormat {
    JSON(1),
    CSV(2),
    STATS(3),
    BINARY(4),
    DIAGNOSTICS(5);

    private final int nativeValue;

    FastParseFormat(int nativeValue) {
        this.nativeValue = nativeValue;
    }

    public int nativeValue() {
        return nativeValue;
    }
}
